gdrdrv/gdrdrv.c

```diff
@@ -872,14 +993,16 @@ static int gdrdrv_mmap(struct file *filp, struct vm_area_struct *vma)
 
 out:
     if (ret) {
-        mr->vma = NULL;
-        mr->mapping = NULL;
-        mr->cpu_mapping_type = GDR_MR_NONE;
-        // TODO: tear down stale partial mappings
+        if (mr) {
+            mr->vma = NULL;
+            mr->mapping = NULL;
+            mr->cpu_mapping_type = GDR_MR_NONE;
+            // TODO: tear down stale partial mappings
```

Comment:
This comment might have never been true, i.e. in case of error, the vma is destroyed and all partial mappings are automatically removed. 
Still we should verify that.

Reply:
Confirmed that if gdrdrv_mmap returns an error, Linux kernel will clean up any partial mappings for us. See https://elixir.bootlin.com/linux/latest/source/mm/mmap.c#L1788 for reference.